cpacker/MemGPT
# feat: add password protection to `/admin` routes on server

**Please describe the purpose of this pull request.**

- Puts `/admin` routes on REST server (used to create users, create API keys, etc) behind bearer token auth by default
- The password (to be passed as a bearer token) can be specified when running the server using env variable `MEMGPT_SERVER_PASS`
- If the password is not specified, it will be auto-generated (and written to stdout)

**How to test**

- Start server
- Try using `/admin` route w/o bearer token, should get `401 Unauthorized`
- Try adding the bearer token, it should then work
- Also try specifying the password via the env var

**Have you tested this PR?**

```sh
 % memgpt server
Generated admin server password for this session: jBNDBB...w5_wg
```

<img width="845" alt="image" src="https://github.com/cpacker/MemGPT/assets/5475622/a47cea0d-6094-4e51-8cb5-f29c977e9fcf">

<img width="840" alt="image" src="https://github.com/cpacker/MemGPT/assets/5475622/67bfce67-9074-475d-ac50-89bdb14ec7bd">

```sh
% export MEMGPT_SERVER_PASS="banana"
% memgpt server                     
Using existing admin server password from environment.
```
<img width="891" alt="image" src="https://github.com/cpacker/MemGPT/assets/5475622/6861bd24-b10d-408f-a17b-2d23017a2083">
